src/Coriolis/non_traditional_beta_plane.jl

```diff
@@ -0,0 +1,80 @@
+"""
+    NonTraditionalBetaPlane{FT} <: AbstractRotation
+
+A Coriolis implementation that accounts for the latitudinal variatino of both
+the locally vertical and the locally horizontal components of the rotation vector.
+Traditionally (see [`BetaPlane`](@ref)) only the locally vertical component is
+accounted for.
+
+This implementation is based off of section 5 of Dellar (2011). It conserve energy,
+angular momentum, and potential vorticity.
+
+References
+==========
+Dellar, P. (2011). Variations on a beta-plane: Derivation of non-traditional
+    beta-plane equations from Hamilton's principle on a sphere. Journal of
+    Fluid Mechanics, 674, 174-195. doi:10.1017/S0022112010006464
+"""
+struct NonTraditionalBetaPlane{FT} <: AbstractRotation
+    fz :: FT
+    fy :: FT
+    β  :: FT
```

Comment:
Would it be better to use the notation `βy` and `βz`?

Reply:
I think so but I chose β and γ to remain consistent with Dellar (2011). I'm indifferent either way but if you think `βy` and `βz` is better I can make the change.

Reply:
Does Dellar (2011) use `fz` and `fy`? There isn't really standard notation for these things: there's also something called the "Gamma-plane" (adding a term proportional to y^2) that involves a parameter called `γ`. I think we should strive to be internally-consistent first, and consistent with outside conventions (where applicable) second. What do you think?

Reply:
Ah sorry I accidently merged yesterday without addressing this point.

Dellar (2011) and other literature I've read commonly use f and f̃ (`\tilde{f}`) to refer to `fz` and `fy`, although I think `fz` and `fy` is clearer and we did decide to switch to `fz` and `fy` at some point. Following the same logic, `βz` and `βy` would be both consistent and clearer.